### DISCUSSION TITLE
Kubernetes 1.25 - kafka upgrade issue due to old strimzi operator

### DISCUSSION THREAD

**Discussion by edgarkz** [strimzi/strimzi-kafka-operator#9119] [Q&A]
Hello,

We have encountered operator\kafka\k8s incompatibility issue and looking for any advise how to proceed from here.
Installed:
**Strimzi operator - 0.27
Kafka - version 2.8.0**

We have recently upgraded to eks **k8s 1.25,** however we missed the strimzi operator api deprecation and now having the following error in the operator log which causes prevents the operator from any reconciliation tasks:

`2023-09-13 20:26:24 ERROR AbstractOperator:247 - Reconciliation #584(timer) Kafka(kafka-data/kafka-data): createOrUpdate failed
io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: POST at: https://10.100.0.1/apis/policy/v1beta1/namespaces/kafka-data/poddisruptionbudgets. Message: the server could not find the requested resource. Received status: Status(apiVersion=v1, code=404, details=StatusDetails(causes=[], group=null, kind=null, name=null, retryAfterSeconds=null, uid=null, additionalProperties={}), kind=Status, message=the server could not find the requested resource, metadata=ListMeta(_continue=null, remainingItemCount=null, resourceVersion=null, selfLink=null, additionalProperties={}), reason=NotFound, status=Failure, additionalProperties={}).
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.requestFailure(OperationSupport.java:697) ~[io.fabric8.kubernetes-client-5.10.2.jar:?]
	 `
We can't downgrade kubernetes.
We can't upgrade operator since current kafka clusters will not be compatible with recent strimzi operator versions.
We can't upgrade kafka to 3.0 due to k8s pdb api removals (0.27 operator is broken atm)

Any ideas how to proceed from here? is there any manual way to upgrade kafka without using the operator so we can make it  to version 3.0 and  install newer operator 0.29 which has builtin support for both k8s api and kafka version?

Thank you

**Comment by scholzj** [comment]
You should be able to upgrade the operator and Kafka in one go. You install the new operator and change the `.spec.kafka.version` field in the Kafka custom resource and can jump directly to Kafka 3.0.0.

**Reply by edgarkz** [reply]
Just to make sure:
I need to change **kafka spec to 3.0** and then upgrade the **operator to 0.29**  - once done - operator will initiate the upgrade process from current 2.8.0 (which is unsupported in new 0.29 operator version based on kafka matrix) to kafka 3.0 ?

Thank you @scholzj

**Reply by scholzj** [reply]
Yes. Correct.

**Reply by edgarkz** [reply]
thank you, that worked!
